NTupleMaker/interface/functionsCP.h

```diff
@@ -795,60 +850,67 @@ if(Pi1[1]==Pi2[1]&&Pi1[1]!=0)cout<<"BIG  strange issue, we're using one vector f
   //  std::cout << "Ux(1) = " << ref1.Px() << "  Uy(1) = " << ref1.Py() << "  Uz(1) = " << ref1.Pz() << std::endl;
   //  std::cout << "Ux(2) = " << ref2.Px() << "  Uy(2) = " << ref2.Py() << "  Uz(2) = " << ref2.Pz() << std::endl;
   //  std::cout << std::endl;
+  
   // get 3-vectors
   TVector3 vecPi1 = Pi1.Vect();
   TVector3 vecPi2 = Pi2.Vect();
   TVector3 vecRef1 = ref1.Vect();
   TVector3 vecRef2 = ref2.Vect();
 
-  // normalize them 
+  // normalize them
+  
   vecPi1 *= 1/vecPi1.Mag();
   vecPi2 *= 1/vecPi2.Mag();
   vecRef1 *= 1/vecRef1.Mag();
   vecRef2 *= 1/vecRef2.Mag();
-
+    
   // transverse components  
   TVector3 vecRef1transv = vecRef1 - vecPi1*(vecPi1*vecRef1);
   TVector3 vecRef2transv = vecRef2 - vecPi2*(vecPi2*vecRef2);
-  
-  /*
-    cout<<"vecRef1transv.Mag() "<<vecRef1transv.Mag() <<endl;
-    cout<<"vecRef2transv.Mag() "<<vecRef2transv.Mag() <<endl;
-    cout<<"vecRef1transv[0] "<<vecRef1transv[0] <<endl;
-    cout<<"vecRef1transv[1] "<<vecRef1transv[1] <<endl;
-    cout<<"vecRef1transv[2] "<<vecRef1transv[2] <<endl;
-    cout<<"vecRef2transv[0] "<<vecRef2transv[0] <<endl;
-    cout<<"vecRef2transv[1] "<<vecRef2transv[1] <<endl;
-    cout<<"vecRef2transv[2] "<<vecRef2transv[2] <<endl;
-  */
 
   vecRef1transv *= 1/vecRef1transv.Mag();
   vecRef2transv *= 1/vecRef2transv.Mag();
 
   double acop = TMath::ACos(vecRef1transv*vecRef2transv);
-  
+  double acoporiginal=acop;
   double sign = vecPi2 * vecRef1transv.Cross(vecRef2transv);
+  double psioriginal =sign;//
+
   if (firstNegative)
     sign = vecPi1 * vecRef2transv.Cross(vecRef1transv);
+  
   if (sign<0) acop = 2.0*TMath::Pi() - acop;
 
   if (y<0) {
     acop = acop + TMath::Pi();
     if (acop>2*TMath::Pi()) {
       acop = acop - 2*TMath::Pi();
     }
-  }  
+  }
 
-  //cout<<"acop before return from function "<<acop<<endl;
-  return acop;
+  //Merijn added a flag for events where sign is bad. in generator level generally ~5%. 
+  if(isinf(sign)) sign=-3;
+  if(sign!=sign) sign=-3;
```

Comment:
The sign variable is the scalar product between a vector of the negative prong and the vectorial product of the two references... If it goes to infinity it means that one of those operations returns nan.
This should be investigated as soon as possible. Please, next time you notice something like this notify both me and Alexei!
The first check should be to see whether the vector pointers are empty or not, so a check at the beginning of the function for this and some counters or cout messages to see which events present this problem.